x-pack/plugins/observability_solution/dataset_quality/public/components/dataset_quality_details/degraded_field_flyout/possible_mitigations/field_limit/increase_field_mapping_limit.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFieldText,
+  EuiFormRow,
+  EuiButton,
+  EuiFieldNumber,
+} from '@elastic/eui';
+import {
+  fieldLimitMitigationApplyButtonText,
+  fieldLimitMitigationCurrentLimitLabelText,
+  fieldLimitMitigationNewLimitButtonText,
+  fieldLimitMitigationNewLimitPlaceholderText,
+} from '../../../../../../common/translations';
+import { useDegradedFields } from '../../../../../hooks';
+
+export function IncreaseFieldMappingLimit({ totalFieldLimit }: { totalFieldLimit: number }) {
+  // Propose the user a 30% increase over the current limit
+  const proposedNewLimit = Math.round(totalFieldLimit * 1.3);
+  const [newFieldLimit, setNewFieldLimit] = useState<number>(proposedNewLimit);
+  const [isInvalid, setIsInvalid] = useState(false);
+  const { updateNewFieldLimit, isSavingNewFieldLimitInProgress } = useDegradedFields();
+
+  const validateNewLimit = (newLimit: string) => {
+    const parsedLimit = parseInt(newLimit, 10);
+    setNewFieldLimit(parsedLimit);
+    if (totalFieldLimit > parsedLimit) {
+      setIsInvalid(true);
+    } else {
+      setIsInvalid(false);
+    }
+  };
```

Comment:
Looking closer at this portion of logic and with regards to the comments on the state machine, I'm wondering if this validation and constrains would be better handled in the flow of the state machine `mitigations` state, where you can prevent side effects and impossible scenarios easily given the nature of the state machine transitions.

Reply:
My idea of State Machine transitions is mostly with server side interactions. If a user does something which involves a server side actions which alters the state, then that success and error goes into state machine, If something is purely client side, and is not triggering anything, i would like to keep it outside state machine like here `change of input`